server/api/reports.py

```diff
@@ -441,6 +446,36 @@ def round_rows(election: Election):
             total_choice_votes.update(non_cvr_choice_vote)
             total_choice_votes.update(cvr_choice_votes)
 
+        if election.audit_type == AuditType.BATCH_COMPARISON:
+            distinct_batches = (
+                Batch.query.join(SampledBatchDraw)
+                .join(Round)
+                .join(Jurisdiction)
+                .filter_by(election_id=election.id)
+                .filter(SampledBatchDraw.round_id == round.id)
+                .all()
+            )
+            num_distinct_batches = len(distinct_batches)
+            num_distinct_ballots = sum(batch.num_ballots for batch in distinct_batches)
+
+            # get the reported votes aggregated across batches
+            reported_results: dict = {choice.id: 0 for choice in contest.choices}
+            for batch in distinct_batches:
+                reported_results_for_batch = (
+                    batch.jurisdiction.batch_tallies[batch.name][contest.id]
```

Comment:
Accessing `batch.jurisdiction` will make a separate query for each batch's jurisdiction. Could be a bit expensive. Often I will profile API endpoints using Heroku/Sentry tools after we deploy them to identify bottlenecks. (Report generation is already slow and should probably be backgrounded)

We might be able to pull out just the reported vote count we need for each batch at the db level if we decide to optimize it.

Reply:
Yeah I was wondering about perf. This data is already accessed/fetched to populate the "batches" section of the csv for batch-comparison audits (where I copied the logic from), but if thats already slow its still not great to add it again.

Reply:
What data do you use to profile?

Reply:
Oh interesting, if it's already loaded once then maybe we can just reuse it? I guess that might take some finagling

To profile, I use Sentry transaction tracing and Heroku database analytics

Reply:
Do you just find some large actual audit that has run to test with though?

Reply:
Both of those tools work by sampling queries from actual usage, so I usually just look at those periodically to see what's slow. It's less that I'll test a specific change that I just deployed.

Load testing specific changes I usually do in my dev env with some large audit test data. But then you don't have as useful profiling tools.

Sorry, I realize my initial comment was a bit misleading about how I use those tools.

Reply:
Ah got it, I guess I wasn't sure if there was a pattern of testing on fake data at times like right now when audits aren't really actively running to help monitor usage